strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/maven/ChecksumMetadataExpirationStrategy.java

```diff
@@ -0,0 +1,83 @@
+package org.carlspring.strongbox.storage.metadata.maven;
+
+import org.carlspring.commons.encryption.EncryptionAlgorithmsEnum;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.storage.metadata.maven.MetadataExpirationStrategy.Decision.*;
+
+@Component
+public class ChecksumMetadataExpirationStrategy
+        implements MetadataExpirationStrategy
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(ChecksumMetadataExpirationStrategy.class);
+
+    @Inject
+    private ProxyRepositoryArtifactResolver proxyRepositoryArtifactResolver;
+
+    public Decision decide(final RepositoryPath repositoryPath) throws IOException
```

Comment:
@cbrianhill I think that you changed the original logic here.

In the currently existing method (`org.carlspring.strongbox.providers.io.MavenMetadataExpiredRepositoryPathHandler#handleExpiration` ) `MD5` case is checked only if `SHA1` case didn't give us a "valid" decision (either `YES_FETCH` or `NO_LEAVE_IT`).

In your case following situation might happen:
* `RepositoryPath checksumRepositoryPath = resolveSiblingChecksum(repositoryPath, EncryptionAlgorithmsEnum.SHA1);`
* `currentChecksum = readChecksum(checksumRepositoryPath);`
* `if (currentChecksum == null)` returns `false`
* `proxyRepositoryArtifactResolver.fetchRemoteResource(checksumRepositoryPath);`
* `final String newRemoteChecksum = readChecksum(checksumRepositoryPath);`
* `if (newRemoteChecksum == null)` returns `true`
* the result is `UNDECIDED;` however we should check `MD5` case in this situation, but your implementation isn't doing it

Reply:
Thank you for pointing this out, @fuss86 .  I had not realized I modified the logic when I attempted to streamline it.